packages/html/src/action-sheet/action-sheet.spec.tsx

```diff
@@ -0,0 +1,123 @@
+import { classNames } from '../utils';
+import { ActionSheetHeader, ActionSheetItems, ActionSheetFooter } from '../action-sheet';
+
+export const ACTIONSHEET_CLASSNAME = `k-actionsheet`;
+
+const states = [];
+
+const options = {};
+
+export type KendoActionSheetProps = {
+    children: JSX.Element | JSX.Element[],
+    title?: string;
+    header?: typeof ActionSheetHeader;
+    footer?: string | typeof ActionSheetFooter;
+    actions?: string[];
+    fullscreen?: boolean;
+    adaptive?: boolean;
+    side?: 'top' | 'right' | 'bottom' | 'left';
+}
+
+const defaultProps = {
+    side: 'bottom',
+    fullscreen: false,
+    adaptive: false
+};
+
+// eslint-disable-next-line complexity
+export const ActionSheet = (
+    props: KendoActionSheetProps &
+        React.HTMLAttributes<HTMLDivElement>
+) => {
+    const {
+        children,
+        title,
+        header,
+        footer,
+        actions,
+        side = defaultProps.side,
+        fullscreen = defaultProps.fullscreen,
+        adaptive = defaultProps.adaptive,
+        ...other
+    } = props;
+
+    // eslint-disable-next-line no-nested-ternary
+    const _ActionSheetHeader = title
+        ? <ActionSheetHeader title={title} />
+        : header
+            ? header
+            : Array.isArray(children) && children.find((child) => child.type === ActionSheetHeader);
+
+    // eslint-disable-next-line no-nested-ternary
+    const _ActionSheetFooter = actions
+        ? <ActionSheetFooter className="k-actions" actions={actions} />
+        // eslint-disable-next-line no-nested-ternary
+        : footer
+            ? typeof footer === 'string'
+                ? <ActionSheetFooter>{footer}</ActionSheetFooter>
+                : footer
+            : Array.isArray(children) && children.find((child) => child.type === ActionSheetFooter);
+
+    // eslint-disable-next-line no-nested-ternary
+    const _ActionSheetContent = Array.isArray(children)
+        ? children.filter(child => {
+            switch (child.type) {
+                case ActionSheetHeader:
+                case ActionSheetFooter:
+                    return false;
+                default:
+                    return true;
+            }
+            return true;
+        })
+        : children.type === ActionSheetItems
+            ? children
+            : <></>;
```

Comment:
Is this based on the current implementation of the `ActionSheet`?

Let's revisit this at a later stage. Overall I would advocate against different rendering modes based on the children or their type (e.g. `string` vs `component`). Seems to be present in multiple components (e.g. List, Page, OrgChart)